src/misc/unicode.cpp

```diff
@@ -1361,49 +1885,73 @@ static bool construct_mapping(const uint16_t code_page)
 
 	// Construct decomposed mapping
-	construct_decomposed(mappings_normalized_by_codepage[code_page],
-	                     mappings_decomposed_by_codepage[code_page]);
+	construct_decomposed(mappings.dos_to_grapheme_normalized,
+	                     mappings.dos_to_grapheme_decomposed);
+
+	// Construct mapping for box-optimized fallback mode
+	construct_box_fallback(mappings.dos_to_grapheme_normalized,
+	                       mappings.box_code_points);
+
+	// Construct upper/lower case mappings
+	construct_case_mapping(uppercase, mappings.dos_to_grapheme_normalized,
+	                       mappings.uppercase);
+	construct_case_mapping(lowercase, mappings.dos_to_grapheme_normalized,
+	                       mappings.lowercase);
 
 	return true;
 }
 
 static void construct_aliases(const uint16_t code_page)
 {
-	assert(!aliases_normalized_by_codepage.count(code_page));
-	assert(!aliases_decomposed_by_codepage.count(code_page));
-	assert(mappings_normalized_by_codepage.count(code_page));
+	auto& mappings = per_code_page_mappings[code_page];
 
-	const auto& mapping      = mappings_normalized_by_codepage[code_page];
-	auto& aliases_normalized = aliases_normalized_by_codepage[code_page];
+	assert(mappings.aliases_normalized.empty());
+	assert(mappings.aliases_decomposed.empty());
 
-	for (const auto& alias : config_aliases) {
-		if (!mapping.count(alias.first) && mapping.count(alias.second) &&
+	const auto& mapping_normalized = mappings.dos_to_grapheme_normalized;
+	auto& aliases_normalized = mappings.aliases_normalized;
+
+	auto add_alias = [&](const alias_t& alias) {
+		if (!mapping_normalized.count(alias.first) &&
+		    mapping_normalized.count(alias.second) &&
 		    !aliases_normalized.count(alias.first)) {
 			aliases_normalized[alias.first] =
-			        mapping.find(alias.second)->second;
+			        mapping_normalized.find(alias.second)->second;
+		}
+	};
+
+	// Construct aliases based on the configuration
+	for (const auto& alias : config_aliases) {
+		add_alias(alias);
+	}
+
+	// Add box drawing aliases
+	for (const auto& alias_group : box_alias_groups) {
+		for (const auto& alias : alias_group) {
+			add_alias(alias);
 		}
 	}
 
 	// Construct decomposed aliases
-	construct_decomposed(aliases_normalized_by_codepage[code_page],
-	                     aliases_decomposed_by_codepage[code_page]);
+	construct_decomposed(mappings.aliases_normalized,
+	                     mappings.aliases_decomposed);
 }
 
 static bool prepare_code_page(const uint16_t code_page)
 {
-	if (mappings_normalized_by_codepage.count(code_page)) {
+	if (per_code_page_mappings.count(code_page)) {
 		return true; // code page already prepared
 	}
 
 	if (!config_mappings.count(code_page) || !construct_mapping(code_page)) {
 		// Unsupported code page or error
-		mappings_normalized_by_codepage.erase(code_page);
-		mappings_decomposed_by_codepage.erase(code_page);
-		mappings_reverse_by_codepage.erase(code_page);
+		per_code_page_mappings.erase(code_page);
```

Comment:
Suggest quantity comparison (count == 0)

Reply:
OK, changed - everywhere in this file.